pkg/tf2pulumi/convert/tf.go

```diff
@@ -1516,6 +1516,16 @@ func convertVariable(sources map[string][]byte, scopes *scopes,
 		modifiedDefault := camelCaseObjectAttributes(variable.Default)
 		blockBody.SetAttributeValue("default", modifiedDefault)
 	}
+
+	if variable.Default.IsNull() && variable.Default.Type() != cty.NilType {
```

Comment:
Can't this just change the condition above? Instead of `if !variable.Default.IsNull()` have `variable.Default.Type() != cty.NilType`. At the very least make this an else if:
```
if !variable.Default.IsNull() {
...
} else if variable.Default.Type() != cty.NilType {
  blockBody.SetAttributeValue("default", cty.NilVal)
}